akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -240,9 +240,11 @@ object Unzip {
   /**
    * Create a new `Unzip`.
    */
+
   def apply[A, B](): Unzip[A, B] = {
     val shape = new FanOutShape2[(A, B), A, B]("Unzip")
-    new Unzip(shape, new UnzipModule(shape, OperationAttributes.name("Unzip")))
+    val f = (t: Tuple2[A, B]) ⇒ (t._1, t._2)
```

Comment:
this function can be preallocated and moved out of this method (private, with types set to `Any`; then upon usage cast it to the right type)